pkg/test/framework/components/echo/util/traffic/generator.go

```diff
@@ -0,0 +1,105 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traffic
+
+import (
+	"fmt"
+	"time"
+
+	"istio.io/istio/pkg/test/framework/components/echo"
+)
+
+const (
+	waitTimeout     = 30 * time.Second
+	defaultInterval = 1 * time.Second
+)
+
+// Config for a traffic Generator.
+type Config struct {
+	// Source of the traffic.
+	Source echo.Instance
+
+	// Options for generating traffic from the Source to the target.
+	Options echo.CallOptions
+
+	// Interval between successive call operations. If not set, defaults to 1 second.
+	Interval time.Duration
+}
+
+// Generator of traffic between echo instances.
+type Generator interface {
+	// Run the Generator and start sending traffic.
+	Run(stop chan struct{})
+
+	// WaitForResult waits for a short while for the Generator to complete
```

Comment:
maybe clarify it waits for in progress requests (if I read the code right)? So if you do `generator.Run(stop); close(stop); g.WaitForResult()` it will likely only serve a single request(?)

Reply:
On that note - do we need a way to ensure we get at least a few requests? Just worried about someone accidentally doing ^. Even the revision test you wrote might suffer from this if the restart is quick. Although I guess if its quick then we have just missed the test - sending more requests after the rollout is done is not useful.